gems/sorbet-runtime/lib/types/props/value_object.rb

```diff
@@ -0,0 +1,28 @@
+# typed: strict
+# frozen_string_literal: true
+
+module T::Props::ValueObject
+  extend T::Sig
+
+  # A struct is equal to another if the classes and all prop definitions (attributes) and values are the same
+  sig {params(other: T::Props::ValueObject).returns(T::Boolean)}
+  def ==(other)
+    return false unless self.class == other.class
+
+    decorator = self.class.decorator
+    return false unless decorator.all_props == other.class.decorator.all_props
+
+    decorator.all_props.all? do |prop|
+      decorator.get(self, prop) == decorator.get(other, prop)
+    end
+  end
```

Comment:
Curious if you thought about adding this so any class mixing this in can also be used as a Hash key?<sup>[1](https://ruby-doc.org/core-3.0.2/Hash.html#class-Hash-label-User-Defined+Hash+Keys)</sup>
```suggestion
  end
  alias eql? ==
```

Reply:
I think it makes sense, it's the same behavior as regular Ruby structs. I added the alias and an additional test for using value objects as hash keys.